### PR TITLE
Feat: override configuration of mca from cma

### PR DESCRIPTION
**Pull request by qiujian16**
1. override configuration first by mca, than install strategy in cma, and finally default configs
2. remove previous configration merge logic in addon-framework

### CODE REVIEW

File: pkg/index/index.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+		objs, err := clusterManagementAddonInformers.Informer().GetIndexer().ByIndex(ClusterManagementAddonByPlacement, fmt.Sprintf("%s/%s", accessor.GetNamespace(), placementName))
+		if err != nil {
+			utilruntime.HandleError(err)
+			return []string{}
+		}
+
+		var keys []string
+		for _, o := range objs {
+			cma := o.(*addonv1alpha1.ClusterManagementAddOn)
+			klog.V(4).Infof("enqueue placeManifestWork %s, because of placementDecision %s/%s",
```
Comment (zhujian7) [inline pkg/index/index.go]: placeManifestWork -> cma?

File: pkg/manager/controllers/addonconfiguration/controller.go [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+	clusterManagementAddonLister  addonlisterv1alpha1.ClusterManagementAddOnLister
+	clusterManagementAddonIndexer cache.Indexer
+
+	reconcilers []addonConfigurationReconcile
+}
+
+// addonConfigurationReconcile is a interface for reconcile logic. It updates configuration of mca based on
+// 1. default configuration on cma
+// 2. configuration on mca
+// 3. configuration in install strategy
```
Comment (skeeey) [inline pkg/manager/controllers/addonconfiguration/controller.go]: nit: we may comment the override order here

File: pkg/manager/controllers/addonconfiguration/addon_configuration_reconciler.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	for _, addon := range graph.addonToUpdate() {
+		mca := d.mergeAddonConfig(addon.mca, addon.desiredConfigs)
+		err := d.patchAddonStatus(ctx, mca, addon.mca)
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if len(errs) > 0 {
+		return cma, reconcileContinue, utilerrors.NewAggregate(errs)
```
Comment (skeeey) [inline pkg/manager/controllers/addonconfiguration/addon_configuration_reconciler.go]: nit: judge the length of errs is not necessary, we can return this directly

**Comment by skeeey** [comment]
In clustermanagement  controller, we also merge the owner refs 
https://github.com/open-cluster-management-io/addon-framework/pull/151/files#diff-4694b02fb2215f3ee9b467379ccca64483dd95c971988ddce3766f1081bda321L105

Did you miss it? or we don't need it any more

**Comment by qiujian16** [comment]
> In clustermanagement controller, we also merge the owner refs https://github.com/open-cluster-management-io/addon-framework/pull/151/files#diff-4694b02fb2215f3ee9b467379ccca64483dd95c971988ddce3766f1081bda321L105
> 
> Did you miss it? or we don't need it any more

right we would need it

**Comment by skeeey** [comment]
LGTM

**Comment by haoqing0110** [comment]
LGTM

File: Makefile [hunk 4]
```diff
@@ -108,25 +113,13 @@ undeploy-helloworld-hosted: ensure-kustomize
...
 	./e2e.test -test.v -ginkgo.v
 
 build-hosted-e2e:
 	go test -c ./test/e2ehosted
 
-test-hosted-e2e: build-hosted-e2e deploy-hosted-ocm deploy-helloworld-hosted
+test-hosted-e2e: build-hosted-e2e deploy-hosted-ocm deploy-addon-manager deploy-helloworld-hosted
 	./e2ehosted.test -test.v -ginkgo.v
 
-# Ensure kustomize
```
Comment (zhujian7) [inline Makefile:120]: nit: why delete this?

Comment (qiujian16) [reply Makefile:120]: we directly use the ensure kustomize target in build-machinery

File: deploy/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+namespace: open-cluster-management-hub
```
Comment (haoqing0110) [inline deploy/kustomization.yaml:1]: Namespace is not created when running `make deploy-addon-manager`. Might needs a `namespace.yaml` under resources.
```
apiVersion: v1
kind: Namespace
metadata:
  name: open-cluster-management-hub
```

**Pull request merged**